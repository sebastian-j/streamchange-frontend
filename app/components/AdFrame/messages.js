/*
 * AdFrame Messages
 *
 * This contains all the text for the AdFrame component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'streamchange.components.AdFrame';

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Advertisement',
  },
  showMoreBtn: {
    id: `${scope}.showMoreButton`,
    defaultMessage: 'Show the details',
  },
  emptyAdText: {
    id: `${scope}.emptyAdPlaceholder`,
    defaultMessage:
      'This is a place for your advertisement. Prices starts from 10 PLN per day. Your ad will reach thousands of viewers.',
  },
  adDialogTitle: {
    id: `${scope}.dialogTitle`,
    defaultMessage: 'Information for advertisers',
  },
  adDialogFirstLine: {
    id: `${scope}.dialogFirstLine`,
    defaultMessage:
      'There are three possible forms of ad: Youtube channel (clickable logo, title, few sentences), Youtube video (embedded video with some text), clickable banner.',
  },
  adDialogSecondLine: {
    id: `${scope}.dialogSecondLine`,
    defaultMessage:
      'If you chose full version, your ad will be displayed on every stream that will show streamchange app that day. Bank transfer is a preferred payment method.',
  },
  adDialogThirdLine: {
    id: `${scope}.dialogThirdLine`,
    defaultMessage: 'Contact:',
  },
  adDialogCloseBtn: {
    id: `${scope}.dialogCloseButton`,
    defaultMessage: 'Close',
  },
});
