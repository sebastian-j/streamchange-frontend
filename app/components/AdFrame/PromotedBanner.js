import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const Slide = keyframes`
  0% {
    transform: translateY(0px);
  }
  50% {
    transform: translateY(-180px);
  }
  100% {
    transform: translateY(0px);
  }
`;

const AdTitle = styled.span`
  color: #7b7b7b;
  font-family: Arial, sans-serif;
  margin-bottom: 5px;
`;

const PromotedContentWrapper = styled.div`
  animation: ${Slide} 240s linear infinite;
  position: relative;
  top: calc(55vh - ${({ offset }) => offset}px);
`;

const Image = styled.img`
  width: 31vw;
`;

const PromotedBanner = props => {
  const [imgHeight, setImgHeight] = useState(0);
  const isVideo = props.imageUrl.substr(props.imageUrl.length - 3) === 'mp4';

  const onImgLoad = ({ target: img }) => {
    setImgHeight(img.offsetHeight);
  };

  const onVideoLoad = ({ target: video }) => {
    setImgHeight(Math.round(video.videoHeight * 0.6));
  };

  return (
    <PromotedContentWrapper offset={imgHeight}>
      <AdTitle>
        <FormattedMessage {...messages.title} />
      </AdTitle>
      <a href={props.channelUrl} target="_blank">
        {!isVideo && (
          <Image alt={props.title} onLoad={onImgLoad} src={props.imageUrl} />
        )}
        {isVideo && (
          <video
            width="100%"
            autoPlay
            loop
            muted
            onLoadedMetadata={onVideoLoad}
          >
            <source src={props.imageUrl} type="video/mp4" />
          </video>
        )}
      </a>
    </PromotedContentWrapper>
  );
};

PromotedBanner.propTypes = {
  channelUrl: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default PromotedBanner;
