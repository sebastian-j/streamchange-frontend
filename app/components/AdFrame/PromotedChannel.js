import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const Slide = keyframes`
  0% {
    top: 33vh;
  }
  50% {
    top: 10vh;
  }
  100% {
    top: 33vh;
  }
`;

const AdTitle = styled.span`
  color: #7b7b7b;
  font-family: Arial, sans-serif;
  margin-bottom: 5px;
`;

const PromotedContentWrapper = styled.div`
  animation: ${Slide} 240s linear infinite;
  position: relative;
`;

const PromotedContent = styled.div`
  border: 1px solid #cecece;
  color: ${(props) => props.theme.staticTextColor};
  display: flex;
  flex-direction: row;
  padding: 14px;
`;

const PromotedChannelLogo = styled.img`
  padding-right: 5px;
  width: 7vw;
`;

const PromotedChannelTitle = styled.a`
  color: ${(props) => props.theme.staticTextColor};
  font-weight: bold;
  font-size: 1.3rem;
  margin-bottom: 4px;
  text-decoration: none;
`;

const PromotedChannel = (props) => (
  <PromotedContentWrapper>
    <AdTitle>
      <FormattedMessage {...messages.title} />
    </AdTitle>
    <PromotedContent>
      <a href={props.channelUrl} target="_blank">
        <PromotedChannelLogo alt={props.title} src={props.imageUrl} />
      </a>
      <div>
        <PromotedChannelTitle href={props.channelUrl} target="_blank">
          {props.title}
        </PromotedChannelTitle>
        <div>{props.description}</div>
      </div>
    </PromotedContent>
  </PromotedContentWrapper>
);

PromotedChannel.propTypes = {
  channelUrl: PropTypes.string.isRequired,
  description: PropTypes.string,
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default PromotedChannel;
