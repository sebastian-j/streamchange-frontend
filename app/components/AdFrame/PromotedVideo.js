import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const Slide = keyframes`
  0% {
    top: 33vh;
  }
  50% {
    top: 10vh;
  }
  100% {
    top: 33vh;
  }
`;

const AdTitle = styled.span`
  color: #7b7b7b;
  font-family: Arial, sans-serif;
  margin-bottom: 5px;
`;

const PromotedContentWrapper = styled.div`
  animation: ${Slide} 240s linear infinite;
  position: relative;
`;

const PromotedContent = styled.div`
  border: 1px solid #cecece;
  color: ${(props) => props.theme.staticTextColor};
  display: flex;
  flex-direction: row;
  padding: 14px;
`;

const PromotedVideo = (props) => (
  <PromotedContentWrapper>
    <AdTitle>
      <FormattedMessage {...messages.title} />
    </AdTitle>
    <PromotedContent>
      <iframe
        id="ytplayer"
        width="500"
        height="170"
        src={`https://www.youtube.com/embed/${props.videoId}?autoplay=0origin=https://www.streamchange.pl`}
        frameBorder="0"
        title="video"
      />
      <div>{props.description}</div>
    </PromotedContent>
  </PromotedContentWrapper>
);

PromotedVideo.propTypes = {
  description: PropTypes.string,
  videoId: PropTypes.string.isRequired,
};

export default PromotedVideo;
