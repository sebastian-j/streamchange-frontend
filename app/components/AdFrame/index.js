import React, { useEffect, useState } from 'react';
import axios from 'axios';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';

import messages from './messages';
import PromotedBanner from './PromotedBanner';
import PromotedChannel from './PromotedChannel';
import PromotedVideo from './PromotedVideo';

const AdPlaceholder = styled.div`
  border: 1px solid #7b7b7b;
  color: #7b7b7b;
  padding: 14px;
  position: relative;
  top: 28vh;
`;

const ColoredLink = styled.a`
  color: ${(props) => props.theme.color};
  display: block;
  padding-top: 5px;
`;

const AdFrame = () => {
  const [promotedContent, setPromotedContent] = useState(null);
  const [isOpen, setIsOpen] = useState(false);

  const openDialog = () => {
    setIsOpen(true);
  };

  const closeDialog = () => {
    setIsOpen(false);
  };

  const getToday = () => {
    const d = new Date();
    let month = `${d.getMonth() + 1}`;
    let day = `${d.getDate()}`;
    const year = d.getFullYear();

    if (month.length < 2) month = `0${month}`;
    if (day.length < 2) day = `0${day}`;

    return [year, month, day].join('-');
  };

  const getPromotedChannels = () => {
    axios.get('../static/sellers.json').then((res) => {
      if (res.data.items) {
        for (let i = 0; i < res.data.items.length; i += 1) {
          if (res.data.items[i].date.includes(getToday()))
            setPromotedContent(res.data.items[i]);
        }
      } else {
        setPromotedContent(null);
      }
    });
  };

  useEffect(() => {
    getPromotedChannels();
  }, []);

  if (promotedContent === null) {
    return (
      <div>
        <AdPlaceholder>
          <span>
            <FormattedMessage {...messages.emptyAdText} />
          </span>
          <Button onClick={openDialog} color="primary">
            <FormattedMessage {...messages.showMoreBtn} />
          </Button>
        </AdPlaceholder>
        <Dialog
          open={isOpen}
          onClose={closeDialog}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">
            <FormattedMessage {...messages.adDialogTitle} />
          </DialogTitle>
          <DialogContent>
            <DialogContentText>
              <FormattedMessage {...messages.adDialogFirstLine} />
            </DialogContentText>
            <DialogContentText>
              <FormattedMessage {...messages.adDialogSecondLine} />
            </DialogContentText>
            <DialogContentText>
              <FormattedMessage {...messages.adDialogThirdLine} />
              <Typography>Discord: Baf#5501</Typography>
              <ColoredLink href="https://www.facebook.com/BafYT/">
                https://www.facebook.com/BafYT/
              </ColoredLink>
              <ColoredLink href="https://www.instagram.com/grafika3d/">
                Instagram grafika3d
              </ColoredLink>
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={closeDialog} color="secondary">
              <FormattedMessage {...messages.adDialogCloseBtn} />
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
  if (promotedContent.kind === 'channel') {
    return (
      <PromotedChannel
        channelUrl={promotedContent.channelUrl}
        description={promotedContent.description}
        imageUrl={promotedContent.imageUrl}
        title={promotedContent.title}
      />
    );
  }
  if (promotedContent.kind === 'image') {
    return (
      <PromotedBanner
        channelUrl={promotedContent.channelUrl}
        imageUrl={promotedContent.imageUrl}
        title={promotedContent.title}
      />
    );
  }
  if (promotedContent.kind === 'video') {
    return (
      <PromotedVideo
        description={promotedContent.description}
        videoId={promotedContent.videoId}
      />
    );
  }
  return <div />;
};

export default AdFrame;
