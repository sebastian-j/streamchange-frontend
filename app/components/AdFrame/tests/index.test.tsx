import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import AdFrame from '../index';

const shallowRenderer = createRenderer();

describe('<AdFrame />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<AdFrame />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
