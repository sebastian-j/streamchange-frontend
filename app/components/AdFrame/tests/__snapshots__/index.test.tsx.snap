// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdFrame /> should render and match the snapshot 1`] = `
<div>
  <styled.div>
    <span>
      <FormattedMessage
        defaultMessage="This is a place for your advertisement. Prices starts from 10 PLN per day. Your ad will reach thousands of viewers."
        id="streamchange.components.AdFrame.emptyAdPlaceholder"
        values={Object {}}
      />
    </span>
    <ForwardRef(Button)
      color="primary"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Show the details"
        id="streamchange.components.AdFrame.showMoreButton"
        values={Object {}}
      />
    </ForwardRef(Button)>
  </styled.div>
  <ForwardRef(Dialog)
    aria-labelledby="form-dialog-title"
    onClose={[Function]}
    open={false}
  >
    <ForwardRef(DialogTitle)
      id="form-dialog-title"
    >
      <FormattedMessage
        defaultMessage="Information for advertisers"
        id="streamchange.components.AdFrame.dialogTitle"
        values={Object {}}
      />
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText)>
        <FormattedMessage
          defaultMessage="There are three possible forms of ad: Youtube channel (clickable logo, title, few sentences), Youtube video (embedded video with some text), clickable banner."
          id="streamchange.components.AdFrame.dialogFirstLine"
          values={Object {}}
        />
      </ForwardRef(DialogContentText)>
      <ForwardRef(DialogContentText)>
        <FormattedMessage
          defaultMessage="If you chose full version, your ad will be displayed on every stream that will show streamchange app that day. Bank transfer is a preferred payment method."
          id="streamchange.components.AdFrame.dialogSecondLine"
          values={Object {}}
        />
        <styled.a
          href="/docs/advertisers-rules"
          target="_blank"
        >
          <FormattedMessage
            defaultMessage="Guidelines for advertisers"
            id="streamchange.components.AdFrame.dialogRulesLink"
            values={Object {}}
          />
          <svg
            focusable="false"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </g>
          </svg>
        </styled.a>
      </ForwardRef(DialogContentText)>
      <ForwardRef(DialogContentText)>
        <FormattedMessage
          defaultMessage="Contact:"
          id="streamchange.components.AdFrame.dialogThirdLine"
          values={Object {}}
        />
        <ForwardRef(Typography)>
          Discord: Baf#5501
        </ForwardRef(Typography)>
        <styled.a
          href="https://www.facebook.com/BafYT/"
        >
          Facebook Baf
          <svg
            focusable="false"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </g>
          </svg>
        </styled.a>
      </ForwardRef(DialogContentText)>
      <ForwardRef(DialogContentText)>
        <styled.a
          href="https://www.instagram.com/grafika3d/"
        >
          Instagram grafika3d
          <svg
            focusable="false"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </g>
          </svg>
        </styled.a>
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        color="secondary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Close"
          id="streamchange.components.AdFrame.dialogCloseButton"
          values={Object {}}
        />
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</div>
`;
